### PR TITLE
Add missing currency in action for transaction requests

### LINKED ISSUE #13725: Missing `currency` in action for transaction requests
## Problem

I want to easily fetch the currency when processing the webhooks like `TransactionChargeRequested`. Currently `action` doesn't have a ` currency` field. It means that I need to additionally fetch `transactions` to get a correct `currency`.

---

## 1. General Assumptions
We will extend the `action` type to contain `currency` field. The same as we have for webhooks like  `TransactionInitializeSession` or `TransactionProcessSession`

### Acceptance Criteria

- [ ] `TransactionChargeRequested` has `action.currency` field 
- [ ] `TransactionRefundRequested` has `action.currency` field
- [ ] `TransactionCancelationRequested` has `action.currency` field

### Subscriptions
 `TransactionChargeRequested`, `TransactionRefundRequested`, `TransactionCancelationRequested` should receive a new field `currency` in `action` structure. 

```graphql
subscription {
  event{
    ... on TransactionChargeRequested{
      action{
        actionType
        amount
        # currency <-- missing
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by AjmalPonneth**
I want to merge this change because it resolves #13725

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/webhook/subscription_types.py [hunk 1]
```diff
@@ -1599,6 +1599,7 @@ class TransactionAction(SubscriptionObjectType, AbstractType):
     amount = PositiveDecimal(
         description="Transaction request amount. Null when action type is VOID.",
     )
+    currency = graphene.String(description="Currency code.")
```
Comment (korycins) [inline saleor/graphql/webhook/subscription_types.py:1602]: Can you add `required=True` and label `ADDED_IN_316` to the description?
Something like:
`currency = graphene.String(description="Currency code." + ADDED_IN_316, required=True)`

**Review by korycins — COMMENTED** [review]
Thanks a lot for your contribution! <3 
I've left one small comment. The rest looks good!

File: saleor/graphql/webhook/subscription_types.py (hunk 1 above)
Comment (AjmalPonneth) [reply saleor/graphql/webhook/subscription_types.py:1602]: Ok. I just forgot about that. Will add and update the PR ASAP.

**Review by fowczarek — APPROVED** [review]

**Comment by AjmalPonneth** [comment]
@korycins Updated.

**Review by aniav — APPROVED** [review]

**Comment by aniav** [comment]
@AjmalPonneth could you check the failing linters and update the PR? 🙏

**Comment by AjmalPonneth** [comment]
> @AjmalPonneth could you check the failing linters and update the PR? 🙏

@aniav Can you please check now?

**Comment by korycins** [comment]
@AjmalPonneth Black is failing on `saleor/graphql/webhook/subscription_types.py`.
Have you tried to use pre-commit? https://docs.saleor.io/docs/3.x/developer/community/contributing#coding-style 
It runs the same linters as the CI, so it should be helpful to find all potential linter failures.

**Comment by AjmalPonneth** [comment]
@korycins I am using Windows on my system. After running the pre-commit hook it shows `Executable sh not found`.

File: saleor/graphql/schema.graphql [hunk 2]
```diff
@@ -32887,6 +32887,13 @@ type TransactionAction @doc(category: "Payments") {
 
   """Transaction request amount. Null when action type is VOID."""
   amount: PositiveDecimal
+
+  """
+  Currency code.
+
```
Comment (korycins) [inline saleor/graphql/schema.graphql:32893]:
```suggestion
  
```

This should solve the CI problem with schema.

**Review by aniav — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Comment by aniav** [comment]
Thank you for your contribution @AjmalPonneth ✨

**Comment by AjmalPonneth** [comment]
Thank you Team Saleor for reviewing my PR. Looking for more contribution in futrue🥳

**Pull request merged**